.github/workflows/main.yml

```diff
@@ -0,0 +1,62 @@
+name: ReFrame CI
+
+on: [push, pull_request]
+
+jobs:
+  unittest:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Generic Unittests
+      run: |
+        ./test_reframe.py
+
+  coverage:
```

Comment:
Couldn't coverage just be part of the `unittests` stage above?

Reply:
The coverage itself yes, but I don't know about the upload to codecov